cpp/src/io/orc/orc.hpp

```diff
@@ -41,6 +41,9 @@ static constexpr uint32_t block_header_size = 3;
 // Seconds from January 1st, 1970 to January 1st, 2015
 static constexpr int64_t orc_utc_epoch = 1420070400;
 
+static constexpr int32_t DEFAULT_MIN_NANOS = 0;
+static constexpr int32_t DEFAULT_MAX_NANOS = 999999;
```

Comment:
Could we document why these are the min and max values? These aren't powers of 2 so I assume it's not integer size based, is it limited by some spec?

Reply:
Why don't we upgrade the type of these constants to `uint32_t`?

Reply:
These do come from specs.
Nanosecond timestamp statistics are stored at millisecond precision + the leftover nanoseconds. This is why the max is 999999, one more and you have a full millisecond.

I can derive the max from `chrono`. Let me know.

Should we use unsigned here? AFAIK we should avoid any uints for any arithmetic.

Reply:
I'm fine using the signed ints, and I don't feel the need to use chrono. Just document that we store as ms + ns so 1e6 ns is the largest you'll ever need.

Reply:
But will we have issue with sign vs unsign comparison?

Reply:
The only place these are used is to compare against `min_ns_remainder`/`max_ns_remainder`, and these are signed.

Reply:
Great.